src/lib/schematics/ng-update/upgrade-rules/package-imports-v8/updateAngularMaterialImportsRule.ts

```diff
@@ -0,0 +1,172 @@
+    // failure saying to import symbols in clause.
+    if (!ts.isNamedImports(declaration.importClause.namedBindings)) {
+      return ctx.addFailureAtNode(declaration, Rule.NO_IMPORT_NAMED_SYMBOLS_FAILURE_STR);
+    }
+
+    // If no symbols are in the named bindings then add failure saying to
+    // import symbols in clause.
+    if (!declaration.importClause.namedBindings.elements.length) {
+      return ctx.addFailureAtNode(declaration, Rule.NO_IMPORT_NAMED_SYMBOLS_FAILURE_STR);
+    }
+
+    // Map of submodule locations to arrays of imported symbols.
+    const importMap = new Map<string, Set<string>>();
+
+    // Determine the subpackage each symbol in the namedBinding comes from.
+    for (const importedSymbol of declaration.importClause.namedBindings.elements) {
+      // Confirm the named import is a symbol that can be looked up.
+      if (!ts.isIdentifier(importedSymbol.name)) {
+        return ctx.addFailureAtNode(
+            importedSymbol, importedSymbol.getFullText(sf) + Rule.SYMBOL_NOT_FOUND_FAILURE_STR);
+      }
+      // Get the original symbol where it is declared upstream.
+      const symbol = getOriginalSymbol(importedSymbol.name, checker);
+
+      // If the symbol can't be found, add failure saying the symbol
+      // can't be found.
+      if (!symbol) {
+        return ctx.addFailureAtNode(
+            importedSymbol, importedSymbol.getFullText(sf) + Rule.SYMBOL_NOT_FOUND_FAILURE_STR);
+      }
+
+      // If the symbol has no declarations, add failure saying the symbol can't
+      // be found.
+      if (!symbol.declarations || !symbol.declarations.length) {
+        return ctx.addFailureAtNode(
+            importedSymbol, importedSymbol.getFullText(sf) + Rule.SYMBOL_NOT_FOUND_FAILURE_STR);
+      }
+
+      // The filename for the source file of the node that contains the
+      // first declaration of the symbol.  All symbol declarations must be
+      // part of a defining node, so parent can be asserted to be defined.
+      const fileName = symbol.declarations[0].parent.getSourceFile().fileName;
+      // File the module the symbol belongs to from a regex match of the
+      // filename. This will always match since only @angular/material symbols
+      // are being looked at.
+      const matched = fileName.match(ANGULAR_MATERIAL_FILEPATH_REGEX)!;
+      // The module name where the symbol is defined e.g. card, dialog.  The
+      // first capture group is contains the module name.
+      const moduleName = `${matched[1]}`;
+      if (importMap.has(moduleName)) {
+        importMap.get(moduleName)!.add(symbol.getName());
+      } else {
+        importMap.set(moduleName, new Set([symbol.getName()]));
+      }
+    }
+    const fix = buildFixerString(importMap);
+
+    // Without a fix to provide, show error message only.
+    if (!fix) {
+      return ctx.addFailureAtNode(declaration.moduleSpecifier, Rule.ONLY_SUBPACKAGE_FAILURE_STR);
```

Comment:
What happens if someone imports `Version` from `@angular/material`. In that case it is valid. Maybe not throw an failure at all? Material identifiers should always either match the Regex or are valid primary-entry point exports right?

Reply:
Additional to this, we are going to move the `Version` symbol to `@angular/material/core`.  We will reexport it as deprecated from `@angular/material` for now and then remove it in the future.

Reply:
👍 This also fixes the issue we had in #15387